ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -150,6 +151,11 @@ data ChainDB m blk = ChainDB {
       -- 'getTipPoint' will return the tip of the immutable DB.
     , getTipPoint        :: STM m (Point blk)
 
+      -- | Get block number of the tip of the chain
+      --
+      -- Will return 'genesisBlockNo' if the database is empty.
+    , getTipBlockNo      :: STM m BlockNo
```

Comment:
Why not just have a single `getTipBlock :: STM m blk`? Then the caller can access anything that is needed.

Reply:
If that is needed, we could, but we don't currently keep blocks in memory.

Reply:
I see; what about `getTipHeader m (Header blk)`?

Reply:
Yes, that would probably be possible, though it would have to return a `Maybe` I suppose.

Reply:
Anyway, either works for me :)

Reply:
`getTipHeader :: STM m (Header blk)` would require more changes: when restarting after a corruption of the VolatileDB, the last `k` blocks/headers might be missing, in which case we don't have any header in memory. To read the tip's header, we would have to read the block/header at the tip of the ImmutableDB from disk (so no `m` instead of `STM m`).

We are already tracking the block number of the tip of the ImmutableDB, so we can easily handle this scenario in `STM m` for just the block number.